### ISSUE TITLE
Format list in comments

### ISSUE THREAD

**Issue by dinosaure** [ocaml-ppx/ocamlformat#403]
Currently, `ocamlformat` compacts list in comments like:

https://github.com/mirage/ocaml-git/pull/311/files#diff-944afc38bfb1daa5cbe9926891267f3dL497

It could be interesting to format lists in comments or keep what the user did.

**Comment by gpetiot** [comment]
Smaller example:
```
val read: state -> Hash.t -> (value, error) result Lwt.t
  (** [read state hash] can retrieve a git {i packed} object from any
      {i PACK} files available in the current git repository
      [state]. It just inflates the git object and informs some
      meta-data (like kind, CRC-32 check-sum, length, etc.) about
      it. Then, the client can use the related decoder to get the
      OCaml value. THIS LINE IS VERY VERY VERY VERY VERY VERY VERY VERY VERY LONG AND SHOULD WRAP
      This function allocates 2 {Cstruct.t} needed to re-construct the
      git {i packed} object in any case (if it is delta-ified or not)
      and a certain amount of little buffer (sized by 0x7F bytes) to
      help the construction only when the requested object is
      delta-ified.

      Can return an {!error}:
      {ul
      {- {!FS.File.error} or {!FS.Dir.error} or
      {!FS.Mapper.error} when we retrieve a file-system error}
      {- {!PACKDecoder.error} when we retrieve an error about the
      decoding of the {i packed} git object in the founded {i PACK}i
      file}
      {- {!IDXDecoder.error} when we retrieve an error about the
      decoding of an {i IDX} file}
      {- [`Not_found] when the requested object is not {i packed}}} *)
```

**Comment by jberdine** [comment]
I am hesitant to start down the path of essentially parsing docstrings / comments. Using an existing/canonical docstring parser, and writing a formatter for its parsed form, sounds like a much more robust and maintainable approach. So my inclination is to close this in favor of #268. Thoughts?

**Comment by gpetiot** [comment]
Closed in favor of #268

**Comment by gpetiot** [comment]
PR #460 should address this issue, I'm interested in getting some feedback
